engine/lib/database.php

```diff
@@ -41,14 +45,15 @@ function execute_delayed_read_query($query, $handler = "") {
  * argument to $callback.  If no callback function is defined, the
  * entire result set is returned as an array.
  *
- * @param mixed  $query    The query being passed.
- * @param string $callback Optionally, the name of a function to call back to on each row
+ * @param string   $query    The query being passed.
+ * @param callable $callback Optionally, the name of a function to call back to on each row
+ * @param array    $params   Query params. E.g. [1, 'steve'] or [':id' => 1, ':name' => 'steve']
  *
  * @return array An array of database result objects or callback function results. If the query
  *               returned nothing, an empty array.
  */
-function get_data($query, $callback = "") {
-	return _elgg_services()->db->getData($query, $callback);
+function get_data($query, $callback = null, array $params = []) {
```

Comment:
Would be nicer to have the params as second param instead of the third. But I guess it doesn't really matter.